src/cluster_undersample/interface_mlj.jl

```diff
@@ -0,0 +1,146 @@
+
+### ClusterUndersampler with MLJ Interface
+# interface struct
+mutable struct ClusterUndersampler{T} <: Static
+    mode::AbstractString
+    ratios::T
+    maxiter::Integer
+    rng::Integer
+    try_perserve_type::Bool
+end;
+
+"""
+Initiate a cluster undersampling model with the given hyper-parameters.
+"""
+function ClusterUndersampler(;
+    mode::AbstractString = "nearest",
+    ratios::Union{Nothing, AbstractFloat, Dict{T, <:AbstractFloat}} = 1.0,
+    maxiter::Integer = 100,
+    rng::Integer = 42,
+    try_perserve_type::Bool = true,
+) where {T}
+    model = ClusterUndersampler(mode, ratios, maxiter, rng, try_perserve_type)
+    return model
+end
+
+"""
+Undersample data X, y 
+"""
+function MMI.transform(r::ClusterUndersampler, _, X, y)
+    return cluster_undersample(
+        X,
+        y;
+        mode = r.mode,
+        ratios = r.ratios,
+        maxiter = r.maxiter,
+        rng = r.rng,
+        try_perserve_type = r.try_perserve_type,
+    )
+end
+
+MMI.metadata_pkg(
+    ClusterUndersampler,
+    name = "Imbalance",
+    package_uuid = "c709b415-507b-45b7-9a3d-1767c89fde68",
+    package_url = "https://github.com/JuliaAI/Imbalance.jl",
+    is_pure_julia = true,
+)
+
+MMI.metadata_model(
+    ClusterUndersampler,
+    input_scitype = Union{Table(Continuous), AbstractMatrix{Continuous}},
+    output_scitype = Union{Table(Continuous), AbstractMatrix{Continuous}},
+    target_scitype = AbstractVector,
+    load_path = "Imbalance." * string(ClusterUndersampler),
```

Comment:
Just checking, does this expand to `load_path = "Imbalance.MLJ.ClusterUndersampler" ?  (MLJTestInterface.jl should catch an error here, so it's probably good)

Reply:
An echo in my brain tells me that it didn't use to work when I would write "Imbalance.ClusterUndersampler". I could never justify why because I thought they were equivalent. Now it makes sense.

Reply:
I nonetheless replaced them all with the explicit equivalent.